tests/contrib/utils/test_sendgrid.py

```diff
@@ -39,17 +36,38 @@ def setUp(self):
         self.bcc = ['foo-bcc@foo.com', 'bar-bcc@bar.com']
         self.expected_mail_data = {
             'content': [{'type': u'text/html', 'value': '<b>Foo</b> bar'}],
-            'personalizations': [
-                {'cc': [{'email': 'foo-cc@foo.com'}, {'email': 'bar-cc@bar.com'}],
-                 'to': [{'email': 'foo@foo.com'}, {'email': 'bar@bar.com'}],
-                 'bcc': [{'email': 'foo-bcc@foo.com'}, {'email': 'bar-bcc@bar.com'}]}],
+            'personalizations': [{
+                'cc': [{'email': 'foo-cc@foo.com'}, {'email': 'bar-cc@bar.com'}],
+                'to': [{'email': 'foo@foo.com'}, {'email': 'bar@bar.com'}],
+                'bcc': [{'email': 'foo-bcc@foo.com'}, {'email': 'bar-bcc@bar.com'}],
+            }],
             'from': {'email': u'foo@bar.com'},
-            'subject': 'sendgrid-send-email unit test'}
+            'subject': 'sendgrid-send-email unit test',
+            'attachments': [{
+                'content': 'dGhpcyBpcyBzb21lIHRlc3QgZGF0YQ==',
+                'content_id': '<test.txt>',
+                'disposition': 'attachment',
+                'filename': 'test.txt',
+                'type': 'text/plain',
+            }],
+            'mail_settings': {},
+        }
 
     # Test the right email is constructed.
     @mock.patch('os.environ.get')
     @mock.patch('airflow.contrib.utils.sendgrid._post_sendgrid_mail')
     def test_send_email_sendgrid_correct_email(self, mock_post, mock_get):
         mock_get.return_value = 'foo@bar.com'
-        send_email(self.to, self.subject, self.html_content, cc=self.cc, bcc=self.bcc)
+
+        tempdir = tempfile.mkdtemp()
```

Comment:
Since we only need one file, can we use [tempfile.NamedTemporaryFile](https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile) in a context instead of dealing with a temporary directory (and removing it)?

```python
with tempfile.NamedTemporaryFile(mode='wt') as f:
    f.write('this is some test data')
    f.flush()
    send_email(..., files=[f.name])
```